dask/dataframe/core.py

```diff
@@ -4461,9 +4461,12 @@ def hash_shard(df, nparts, split_out_setup=None, split_out_setup_kwargs=None):
         h = split_out_setup(df, **(split_out_setup_kwargs or {}))
     else:
         h = df
-    h = hash_pandas_object(h, index=False)
+    from .utils import hash_object_dispatch
+
+    hash_object_func = hash_object_dispatch.dispatch(type(h))
+    h = hash_object_func(h, index=False)
     if is_series_like(h):
-        h = h._values
+        h = h.values
```

Comment:
This will convert a cuDF GPU object to a numpy array here as of now, is there a reason we need `h` to be a numpy array-like as opposed to leaving it as a Series?

Reply:
Both work in cudf. In pandas modulo works but iloc doesn't yet seem to work in latest pandas(`0.25`) too:
```python
>>> {i: df.iloc[x == i] for i in range(4)}
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 1, in <dictcomp>
  File "/home/nfs/pgali/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py", line 1410, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File "/home/nfs/pgali/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py", line 2118, in _getitem_axis
    self._validate_key(key, axis)
  File "/home/nfs/pgali/anaconda3/lib/python3.7/site-packages/pandas/core/indexing.py", line 1976, in _validate_key
    "iLocation based boolean "
NotImplementedError: iLocation based boolean indexing on an integer type is not available
```